gateway/src/shard/processor/impl.rs

```diff
@@ -285,11 +256,15 @@ impl ShardProcessor {
                             source,
                         })?;
                     let msg_or_error = match decompressed_msg {
-                        Some(json) => serde_json::from_slice(json).map_err(|source| {
-                            Error::PayloadSerialization {
-                                source,
-                            }
-                        }),
+                        Some(json) => {
+                            emit::bytes(self.listeners.clone(), json).await;
```

Comment:
Can this not end up being quite expensive as it may be done many times every second?

Reply:
I don't think so, there's not actually going to really be any contention around the async listeners mutex, so acquiring it on the first poll will probably always be instant. Since the raw payload event type is disabled by default, the bytes are (unless opted into) never going to be cloned, so it'll be a fast operation